client-guide/jmap-client-guide.mdown

```diff
@@ -613,43 +686,96 @@ When you next do resync, whether that's a second or several days later, apply th
-            sort: [ "date desc", "id desc" ]
-            collapseThreads: false,
-            position: 0,
-            limit: 100
-        }, "call3" ]
-    ]
-
-The first two calls get the delta updates to mailboxes and messages (if the client has a copy of all messages locally, it has no need to use the server to get threads as it already has all the information). In the common case where there are fewer than 50 message changes since last time, this will bring the client fully up to date (barring downloading new message bodies, attachments and other details, which it can easily schedule at its leisure).
-
-However, if there have been a large number of changes since last time, the client is not yet fully up to date. It can continue to call getMessageUpdates to fetch more changes until it reaches the current state. Simultaneously, it can use the response to the getMessageList call in the first request to see if there are any new message ids at the top of the Inbox. These can be downloaded first so the user has immediate access to new messages (by the end of the second round trip) while the other changes then continue to sync across afterwards.
+```json
+[
+    // Sync mailbox changes
+    [ "Mailbox/changes", {
+        "accountId": "u123456",
+        "sinceState": "m123456789",
+        "maxChanges": 50
+    }, "0" ],
+    [ "Mailbox/get", {
+        "accountId": "u123456",
+        "#ids": {
+            "name": "Mailbox/changes",
+            "path": "/added",
+            "resultOf": "0"
+        }
+    }, "1" ],
+    [ "Mailbox/get", {
+        "accountId": "u123456",
+        "#ids": {
+            "name": "Mailbox/changes",
+            "path": "/updated",
+            "resultOf": "0"
+        },
+        "#properties": {
+            "name": "Mailbox/changes",
+            "path": "/updatedProperties",
+            "resultOf": "0"
+        }
+    }, "2" ],
+
+    // Sync message changes
+    [ "Email/changes", {
+        "accountId": "u123456",
+        "sinceState": "m815034",
+        "maxChanges": 50
+    }, "3" ],
+    [ "Email/get", {
+        "accountId": "u123456",
+        "#ids": {
+            "name": "Email/changes",
+            "path": "/added",
+            "resultOf": "3"
+        },
+        "properties": [
+            "id",
+            "threadId",
+            "mailboxIds",
+            "keywords",
+            "hasAttachment",
+            "from",
+            "to",
+            "subject",
+            "receivedAt",
+            "preview"
+        ]
+    }, "4" ],
+    [ "Email/get", {
+        "accountId": "u123456",
+        "#ids": {
+            "name": "Email/changes",
+            "path": "/updated",
+            "resultOf": "3"
+        },
+        "properties": [
+            "id",
+            "threadId",
+            "mailboxIds",
+            "keywords",
+            "hasAttachment",
+            "from",
+            "to",
+            "subject",
+            "receivedAt",
+            "preview"
```

Comment:
For updated, you only need to refetch mutable properties: keywords and mailboxIds. The rest of the properties are immutable so can't have changed.